source/documentation/guides/dependencies.html.md.erb

```diff
@@ -0,0 +1,84 @@
+
+The number of open issues and how they are responded to can also be an indicator of community health. It's worth being wary of anything with a high number of "won't fix" type labels.
+
+### 4. Check the license on the dependency
+
+Not all software licenses are compatible. We use an MIT license for our open source code, so anything using that is 100% compatible. 
+
+Some licenses may be incompatible with private storage of source code, for example due to forcing publication, which might not be appropriate to a service that requires some parts of its codebase to be private for operational reasons. 
+
+### 5. Check how the library responded to any previous security incidents
+
+Transparency and quick reaction to any previous security issues are a good sign of how a library's maintainers will react next time. Particularly if they have defined incident processes, public lessons learned and risk mitigations already in place. 
+
+### 6. See how easy contributing is
+
+We like to give back if we use things, hopefully that's not a painful process. How easy is it for a new contributor to get a PR merged on the library? This will help if we do need to patch something upstream to gain compatibility with recent language versions.
+
+This can be a downside too: what checks do you need to pass to get code added? Is it too easy? Could somebody manipulate that process?
+
+### 7. Do you need the whole thing?
+
+Smaller libraries are prefered to larger frameworks to avoid bringing in large amounts of code you don't need for one class or method you do. The attack surface of a bigger project means there's more potential for you to get an exploit in unrelated code you end up shipping anyway.
+
+This is particularly an issue with frontend dependencies, don't add 1 library with 40 sub dependencies just to do something browsers can do anyway with CSS or native APIs. The performance and usability impact of 25mb of JS just to add a hover effect should not be underestimated. Check out tools like tree-shaking in builds that can reduce this by not deploying unused codepaths.
+
+### 8. Check what the library's own dependencies look like
+
+Sometimes it's not the direct dependency itself that's the risk, but the dependencies of the dependency. It's worth checking for further downstream projects with the same set of issues.
+
+### 9. Pin to a specific version
+
+It's always best to pin a given dependency to a specific version. So using 1.0.4 rather than a less specific ^1.0. 
+
+While lock files will pin to known good versions, you might inadvertantly pull in a malicious version 1.0.5 as it becomes available as part of development changes when updating packages if you aren't pinned. 
+
+We have dependabot to pull in point-version updates when they become available. For particularly volatile ecosystems it is best to wait a 3 days before merging PRs for non-security related updates to allow time for malicious activity to be detected.
```

Comment:
We are currently looking at the idea of auto merging depenabot PRs in our pipeline. See this link

https://github.com/ministryofjustice/opg-pdf-service/pull/218

I wonder is it worth exploring how to add a delay of 3 days to anything that auto merges PRs? 

Or are we safer getting merging due to any bug fixes coming out and fixing them quickly? My gut says no.

Reply:
We had auto-merging on Sirius for a bit, but turned it off after a few really bad incidents in the node ecosystem. 3 days felt like enough to catch most malicious activity. There's probably a bit more info on security vs general version bumps that I haven't made clear. So I'll do some clarifying.